### PR TITLE
[WFCORE-4562] Update the pre-recorded messages in CertificateAuthorit…

### PR DESCRIPTION
**Pull request by ashley-abdelsayed98**
…iesTestCase and KeyStoresTestCase using the latest Boulder instance

https://issues.jboss.org/browse/WFCORE-4562

### CODE REVIEW

**Comment by bstansberry** [comment]
this is ok to test

**Comment by bstansberry** [comment]
@fjuma Please review.

File: elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java [hunk 1]
```diff
@@ -1689,217 +1676,217 @@ private ClientAndServer setupTestObtainCertificateWithECPublicKey() {
...
 
 
         final String QUERY_ACCT_RESPONSE_BODY_1= "";
 
-        final String QUERY_ACCT_REPLAY_NONCE_1 = "MW2S_i5GD86wRqzEl1Y8g5rm5vwhypO52NAnRtzW-yg";
+        final String QUERY_ACCT_REPLAY_NONCE_1 = "taroQHBvufOONuGk_QUhYXWLU7X1ua61jh3f9y6HXkoOy1I";
 
-        final String QUERY_ACCT_REQUEST_BODY_2 = "{\"protected\":\"eyJhbGciOiJSUzI1NiIsImtpZCI6Imh0dHA6Ly9sb2NhbGhvc3Q6NDAwMS9hY21lL2FjY3QvMyIsIm5vbmNlIjoiTVcyU19pNUdEODZ3UnF6RWwxWThnNXJtNXZ3aHlwTzUyTkFuUnR6Vy15ZyIsInVybCI6Imh0dHA6Ly9sb2NhbGhvc3Q6NDAwMS9hY21lL2FjY3QvMyJ9\",\"payload\":\"eyJ0ZXJtc09mU2VydmljZUFncmVlZCI6dHJ1ZSwiY29udGFjdCI6WyJtYWlsdG86YWRtaW5AbXlleGFtcGxlLmNvbSJdfQ\",\"signature\":\"VPnJ5ySaCMrLlB2yU71MUZKA_eFptuLbYxnRXs1lRxtPbV79Ql9853ykwENN3Uy-jMqnmMph2Wi_yJRa30-TvvwthY5lxQQzJ-sTVP683JNKPJbjpj7lEMXhfsGhzpScynN4pcFGQtUgYRQ9Yi7e8VSucLorDnyU6_AecY4auELAfAdJy9dDfYTsJCs_u0hC25ob6RS2iIgJTQ3WcC5Y52i8_d8vWQweHD_G0tM-XZWw1foyt_nUpSDbbo9kfRBO4qAL00jfO2av35WF0p35A0iowosjFXtn7Pft9NKDbJzgK5_Ax83mDaKp5dJzcAh_QjQ2TuNU9EMosqOHvi0PbQ\"}";
+        final String QUERY_ACCT_REQUEST_BODY_2 = "";
```
Comment (fjuma) [inline elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java]: Missing the updated value here.

File: elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java [hunk 2]
```diff
@@ -1037,35 +1038,35 @@ public void testObtainCertificateWithoutAgreeingToTermsOfService() throws Except
...
         assertFalse(keyStore.containsAlias(alias));
-        obtainCertificate(keyAlgorithmName, keySize, "mndelkdnbcilohg.com", alias, keyStore);
+        obtainCertificate(keyAlgorithmName, keySize, "mndelkdnbcilohgv2.com", alias, keyStore);
     }
 
     @Test
     public void testObtainCertificateWithUnsupportedPublicKey() throws Exception {
         addKeyStore(ACCOUNTS_KEYSTORE, ACCOUNTS_KEYSTORE_NAME, ACCOUNTS_KEYSTORE_PASSWORD);
-        addCertificateAuthorityAccount("account7");
+        addCertificateAuthorityAccount("account7v2", "mailto:admin@anexample.com");
```
Comment (fjuma) [inline elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java]: Just noticed that the contact url used for "account7v2" is different in the previous test. Is there a reason for this?

Comment (ashley-abdelsayed98) [reply elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java]: Sorry I missed this as the test was mean to fail, updated to the correct email

**Comment by ashley-abdelsayed98** [comment]
Thanks @fjuma I've updated the ```elytron/src/test/resources/org/wildfly/extension/elytron/Readme.txt```

File: elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java (hunk 1 above)
Comment (ashley-abdelsayed98) [reply elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java]: Fixed

**Comment by ashley-abdelsayed98** [comment]
retest this please

**Comment by marekkopecky** [comment]
retest this please

**Comment by marekkopecky** [comment]
@fjuma @ashley-abdelsayed98 What is the status of this PR? Can this PR be merged?

**Comment by fjuma** [comment]
Just one more change needed and then this should be ready to go - the use of ```mockCertificateAuthorityUrl``` in both test classes can now be removed since it's now possible to configure a custom certificate authority URL directly.

**Comment by ashley-abdelsayed98** [comment]
@fjuma Thanks, I've made those changes to use the directly configured custom certificate authority URLs

**Comment by marekkopecky** [comment]
retest this please

**Comment by ashley-abdelsayed98** [comment]
retest this please

File: elytron/src/test/java/org/wildfly/extension/elytron/CertificateAuthoritiesTestCase.java [hunk 3]
```diff
@@ -263,7 +222,8 @@ public void testCreateAccountWithEmptyStagingUrlAndStagingValueTrue() throws Exc
             operation.get(ClientConstants.OP).set(ElytronDescriptionConstants.CREATE_ACCOUNT);
             operation.get(ElytronDescriptionConstants.AGREE_TO_TERMS_OF_SERVICE).set(true);
             operation.get(ElytronDescriptionConstants.STAGING).set(true);
-            assertFailed(services.executeOperation(operation));
```
Comment (fjuma) [inline elytron/src/test/java/org/wildfly/extension/elytron/CertificateAuthoritiesTestCase.java]: I think we should keep this assertion to make sure the operation really did fail before attempting to get the failure description.

File: elytron/src/test/java/org/wildfly/extension/elytron/CertificateAuthoritiesTestCase.java [hunk 4]
```diff
@@ -310,17 +273,20 @@ public void testCreateAccountWithoutAgreeingToTermsOfService() throws Exception
             operation.get(ClientConstants.OP_ADDR).add("subsystem", "elytron").add("certificate-authority-account", CERTIFICATE_AUTHORITY_ACCOUNT_NAME);
             operation.get(ClientConstants.OP).set(ElytronDescriptionConstants.CREATE_ACCOUNT);
             operation.get(ElytronDescriptionConstants.AGREE_TO_TERMS_OF_SERVICE).set(false);
-            assertFailed(services.executeOperation(operation));
```
Comment (fjuma) [inline elytron/src/test/java/org/wildfly/extension/elytron/CertificateAuthoritiesTestCase.java]: Same here.

File: elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java [hunk 5]
```diff
@@ -1078,9 +1045,11 @@ public void testObtainCertificateWithUnsupportedPublicKey() throws Exception {
             operation.get(ElytronDescriptionConstants.KEY_SIZE).set(2048);
             operation.get(ElytronDescriptionConstants.AGREE_TO_TERMS_OF_SERVICE).set(true);
             operation.get(ElytronDescriptionConstants.CERTIFICATE_AUTHORITY_ACCOUNT).set(CERTIFICATE_AUTHORITY_ACCOUNT_NAME);
-            assertFailed(services.executeOperation(operation));
```
Comment (fjuma) [inline elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java]: Same here.

File: elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java [hunk 6]
```diff
@@ -1026,9 +988,11 @@ public void testObtainCertificateWithoutAgreeingToTermsOfService() throws Except
             operation.get(ElytronDescriptionConstants.ALIAS).set(alias);
             operation.get(ElytronDescriptionConstants.DOMAIN_NAMES).add("www.example.com");
             operation.get(ElytronDescriptionConstants.CERTIFICATE_AUTHORITY_ACCOUNT).set(CERTIFICATE_AUTHORITY_ACCOUNT_NAME);
-            assertFailed(services.executeOperation(operation));
```
Comment (fjuma) [inline elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java]: Same here.

File: elytron/src/test/java/org/wildfly/extension/elytron/CertificateAuthoritiesTestCase.java (hunk 3 above)
Comment (ashley-abdelsayed98) [reply elytron/src/test/java/org/wildfly/extension/elytron/CertificateAuthoritiesTestCase.java]: Fixed for all

**Comment by ashley-abdelsayed98** [comment]
retest this please

**Comment by ashley-abdelsayed98** [comment]
retest this please

**Comment by marekkopecky** [comment]
retest this please

**Comment by ashley-abdelsayed98** [comment]
retest this please

**Comment by ashley-abdelsayed98** [comment]
Just fyi this is ready to be merged, thanks!

**Pull request merged**